mambaTests/Util Tests/Value Types/HLSChannelsTests.swift

```diff
@@ -0,0 +1,134 @@
+        let actualChannels = HLSChannels(string: twelveChannelJoc)
+        let expectedChannels = HLSChannels(
+            count: 12,
+            spatialAudioCodingIdentifiers: ["JOC"],
+            specialUsageIdentifiers: []
+        )
+        XCTAssertEqual(expectedChannels, actualChannels)
+    }
+
+    func test_twelveChannelJocAndUnknownSpatialCoding() {
+        let actualChannels = HLSChannels(string: twelveChannelJocAndUnknownSpatialCoding)
+        let expectedChannels = HLSChannels(
+            count: 12,
+            spatialAudioCodingIdentifiers: ["JOC", "SPECIAL"],
+            specialUsageIdentifiers: []
+        )
+        XCTAssertEqual(expectedChannels, actualChannels)
+    }
+
+    func test_sixChannelWithEmptySpatialIdentifier() {
+        let actualChannels = HLSChannels(string: sixChannelWithEmptySpatialIdentifier)
+        let expectedChannels = HLSChannels(
+            count: 6,
+            spatialAudioCodingIdentifiers: [],
+            specialUsageIdentifiers: []
+        )
+        XCTAssertEqual(expectedChannels, actualChannels)
+    }
+
+    func test_twelveChannelUnknownSpatialWithDashInName() {
+        let actualChannels = HLSChannels(string: twelveChannelUnknownSpatialWithDashInName)
+        let expectedChannels = HLSChannels(
+            count: 12,
+            spatialAudioCodingIdentifiers: ["VERY-SPATIAL"],
+            specialUsageIdentifiers: []
+        )
+        XCTAssertEqual(expectedChannels, actualChannels)
+    }
+
+    func test_sixChannelNoSpatialWithDownmix() {
+        let actualChannels = HLSChannels(string: sixChannelNoSpatialWithDownmix)
+        let expectedChannels = HLSChannels(
+            count: 6,
+            spatialAudioCodingIdentifiers: [],
+            specialUsageIdentifiers: [.downmix]
+        )
+        XCTAssertEqual(expectedChannels, actualChannels)
+    }
+
+    func test_sixChannelNoSpatialWithBinauralAndImmersive() {
+        let actualChannels = HLSChannels(string: sixChannelNoSpatialWithBinauralAndImmersive)
+        let expectedChannels = HLSChannels(
+            count: 6,
+            spatialAudioCodingIdentifiers: [],
+            specialUsageIdentifiers: [.binaural, .immersive]
+        )
+        XCTAssertEqual(expectedChannels, actualChannels)
+    }
+
+    func test_twelveChannelJocAndImmersive() {
+        let actualChannels = HLSChannels(string: twelveChannelJocAndImmersive)
+        let expectedChannels = HLSChannels(
+            count: 12,
+            spatialAudioCodingIdentifiers: ["JOC"],
+            specialUsageIdentifiers: [.immersive]
+        )
+        XCTAssertEqual(expectedChannels, actualChannels)
+    }
+
+    // In the case that we don't recognize the special usage identifier, I think it is better to fail parsing the entire
+    // CHANNELS attribute, as otherwise we risk misleading the user of the library into thinking that the special usage
+    // is less than it actually is.
+    func test_sixChannelUnknownSpecialUsageIdentifier() {
```

Comment:
This test is failing locally for me. Returning non-nil

Reply:
actually never mind, I think I had a local change that I forgot to stash. Its passing now